src/test/ui/issues/issue-76191.stderr

```diff
@@ -0,0 +1,22 @@
+error[E0308]: mismatched types
+  --> $DIR/issue-76191.rs:10:9
+   |
+LL | const RANGE: RangeInclusive<i32> = 0..=255;
+   | ------------------------------------------- constant defined here
+...
+LL |     match n {
+   |           - this expression has type `i32`
+LL |         RANGE => {}
+   |         ^^^^^
+   |         |
+   |         expected `i32`, found struct `std::ops::RangeInclusive`
+   |         `RANGE` is interpreted as a constant, not a new binding
+   |         help: introduce a new binding instead: `other_range`
```

Comment:
This one suggestion shouldn't be displayed when we have identified for sure that this is a Range `const`.